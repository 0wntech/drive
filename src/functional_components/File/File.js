import React from 'react';
import styles from './File.module.css';
import {Menu, Separator, MenuProvider, Item} from 'react-contexify';
export default function File({
    currPath,
    onClick,
    image,
    label,
    selectedItem,
    onDelete,
    onAccess,
    onRename,
    onInfo,
}) {
    const imageTypes = ['ico', 'png', 'jpeg', 'jpg'];
    const isImage = imageTypes.indexOf(label.split('.')[1]) > -1;
    console.log(currPath + encodeURIComponent(label));
    if (isImage) {
        return (
            <div>
                <MenuProvider id={label + 'contextmenu'}>
                    <div
                        className={styles.container}
                        style={selectedItem ? {opacity: 0.5} : undefined}
                        onClick={onClick}
                    >
                        <div className={styles.innerContainer}>
                            <div className={styles.iconContainer}>
                                <img
                                    className={styles.thumbnail}
                                    src={currPath + encodeURIComponent(label)}
                                />
                                <img className={styles.icon} src={image} />
                            </div>
                            <p>{label}</p>
                        </div>
                    </div>
                    <Menu id={label + 'contextmenu'}>
                        <Item
                            onClick={() => {
                                onDelete(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Delete</div>
                        </Item>
                        <Item
                            onClick={() => {
                                onInfo(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Info</div>
                        </Item>
                        <Item
                            onClick={() => {
                                onAccess(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Access</div>
                        </Item>
                        <Item
                            onClick={() => {
                                onRename(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Rename</div>
                        </Item>
                        <Separator />
                        <Item
                            onClick={() => {
                                onRename(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Rename</div>
                        </Item>
                    </Menu>
                </MenuProvider>
            </div>
        );
    } else {
        return (
            <div>
                <MenuProvider id={label + 'contextmenu1'}>
                    <div
                        className={styles.container}
                        style={selectedItem ? {opacity: 0.5} : undefined}
                        onClick={onClick}
                    >
                        <div className={styles.innerContainer}>
                            <img className={styles.icon} src={image} />
                            <p className={styles.label}>{label}</p>
                        </div>
                    </div>
                    <Menu id={label + 'contextmenu1'}>
                        <Item
                            onClick={() => {
                                onDelete(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Delete</div>
                        </Item>
                        <Item
                            onClick={() => {
                                onInfo(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Info</div>
                        </Item>
                        <Item
                            onClick={() => {
                                onAccess(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Access</div>
                        </Item>
                        <Item
                            onClick={() => {
                                onRename(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Rename</div>
                        </Item>
                        <Separator />
                        <Item
                            onClick={() => {
                                onRename(currPath + encodeURIComponent(label));
                            }}
                        >
                            <div className={styles.contextItem}>Rename</div>
                        </Item>
                    </Menu>
                </MenuProvider>
            </div>
        );
    }
}
