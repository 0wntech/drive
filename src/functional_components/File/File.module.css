.container {
    cursor: pointer;
    position: relative;
    width: 100%;
    height: 150px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.innerContainer {
    position: absolute;
    width: 100%;
}

.label {
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    text-align: center;
    transition: all 0.2s;
}

.container:hover img {
    opacity: 0.6;
}

.container:hover .innerContainer {
    width: 110%;
    height: 110%;
}

.container:hover .label {
    overflow: visible;
    font-size: 110%;
}

.iconContainer {
    position: relative;
    display: inline;
    max-width: 95%;
}

.thumbnail {
    max-width: 50%;
    height: auto;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
}

.icon {
    height: auto;
    max-width: 100%;
}
