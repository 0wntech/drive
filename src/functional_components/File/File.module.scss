@import '../../../styles/constants.scss';

.container {
    z-index: 1;
    cursor: pointer;
    position: relative;
    width: 80px;
    height: 115px;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
}

.innerContainer {
    position: absolute;
    width: 100%;
}

.fileType {
    z-index: 2;
    position: absolute;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
    font-weight: 300;
    font-size: 1.5em;
    color: #666;
}

.gridCell {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 100%;
    width: 100%;
    position: relative;
}

.selected {
    box-shadow: 0 0 0 1px #6db7db;

    border-radius: 20px;
}

.label {
    font-family: 'Helvetica Neue Light';
    font-style: normal;
    font-weight: 300;
    font-size: 16px;
    color: #000000;
    line-height: 21px;
    letter-spacing: 0.05em;
    z-index: 1;
    width: 90%;
    text-align: center;
    position: absolute;
    background-color: white;
    text-overflow: ellipsis;
    white-space: nowrap;
    overflow: hidden;
    bottom: -1.2em;
    left: 50%;
    transform: translateX(-50%);
    transition: all 0.2s;
    padding: 0 0.5em;
}

.contextMenu {
    background-color: white;
    border-radius: 5px;
    border: 0.5px solid #ddd;
    z-index: 2;
    width: 130px;
    padding: 0 !important;
}

.contextItem div {
    color: #000000;
}

.contextItem:hover div {
    background-color: #f2f2f2 !important;
}

.disabled div {
    color: #8c8c8c;
    cursor: default;
}

.itemLabel {
    z-index: 3;
    color: #000;
}

.gridCell:hover .label {
    overflow: visible;
    font-size: 105%;
}

.container:hover .label {
    overflow: visible;
    font-size: 105%;
}
.icon:hover .label {
    overflow: visible;
    font-size: 105%;
}

.innerContainer:hover .label {
    overflow: visible;
    font-size: 105%;
}

.iconContainer {
    position: relative;
    display: inline;
}

.thumbnail {
    max-width: 50%;
    height: auto;
    position: absolute;
    top: 50%;
    left: 50%;
    z-index: 1;
    transform: translateX(-50%) translateY(-40%);
}

.linkedFile {
    width: 40%;
    position: absolute;
    z-index: 2;
    top: 50%;
    left: 50%;
    transform: translateX(-50%) translateY(-50%);
}

.tooltip {
    position: absolute;
    z-index: 2;
    width: 200%;
    left: 50%;
    transform: translateX(-50%) translateY(-110%);
    top: 0%;
    border: solid #ddd 2px;
    border-radius: 10px;
    display: none;
}

.active {
    display: initial;
}

.tooltip p {
    font-size: 10px;
    text-align: center;
    margin: initial;
    color: #666;
    padding: 0.2em;
}

.icon {
    height: auto;
    max-width: 100%;
    position: relative;
    left: 50%;
    transform: translateX(-50%);
}

@media (max-width: $screen_s) {
    .icon {
        height: 80px;
        max-width: 60px;
    }

    .innerContainer {
        position: initial;
        width: initial;
    }

    .container {
        width: initial;
        height: initial;
    }

    .label {
        line-height: initial;
        position: initial;
        transform: initial;
    }

    .gridCell {
        justify-content: initial;
        height: initial;
        justify-content: space-between;
        height: 112px;
    }
}
