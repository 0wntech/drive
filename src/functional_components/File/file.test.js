import React from 'react';
import TestRenderer from 'react-test-renderer';
import File from './File';
import { shallow } from 'enzyme';

describe('File', () => {
    test('should render correctly (no image)', () => {
        const currPath = 'https://bejow.solid.community/1234/';
        const onClick = jest.fn();
        const image = 'https://owntech.de/favicon.ico';
        const label = 'Test';
        const selectedItem = false;
        const contextMenuOptions = [
            {
                label: 'test',
                onClick: () => {},
                disabled: false,
            },
            {
                label: 'test disabled',
                onClick: () => {},
                disabled: true,
            },
        ];
        const tree = TestRenderer.create(
            <File
                currPath={currPath}
                onClick={onClick}
                image={image}
                label={label}
                selectedItem={selectedItem}
                contextMenuOptions={contextMenuOptions}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
    test('should render correctly (image)', () => {
        const currPath = 'https://bejow.solid.community/1234/';
        const onClick = jest.fn();
        const image = 'https://owntech.de/favicon.ico';
        const label = 'Test.png';
        const selectedItem = false;
        const contextMenuOptions = [
            {
                label: 'test',
                onClick: () => {},
                disabled: false,
            },
            {
                label: 'test disabled',
                onClick: () => {},
                disabled: true,
            },
        ];
        const tree = TestRenderer.create(
            <File
                currPath={currPath}
                onClick={onClick}
                image={image}
                label={label}
                selectedItem={selectedItem}
                contextMenuOptions={contextMenuOptions}
            />
        );
        expect(tree.toJSON()).toMatchSnapshot();
    });
    test('Should have slected class if File is selected', () => {
        const currPath = 'https://bejow.solid.community/1234/';
        const onClick = jest.fn();
        const image = 'https://owntech.de/favicon.ico';
        const label = 'Test.png';
        const selectedItem = true;
        const contextMenuOptions = [
            {
                label: 'test',
                onClick: () => {},
                disabled: false,
            },
            {
                label: 'test disabled',
                onClick: () => {},
                disabled: true,
            },
        ];
        const wrapper = shallow(
            <File
                currPath={currPath}
                onClick={onClick}
                image={image}
                label={label}
                selectedItem={selectedItem}
                contextMenuOptions={contextMenuOptions}
            />
        );
        expect(wrapper.find('.selected').length).toBe(1);
    });
});
