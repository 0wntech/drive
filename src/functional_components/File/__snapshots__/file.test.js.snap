// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`File should render correctly (image) 1`] = `
<div
  className="gridCell"
  onContextMenu={[Function]}
>
  <div
    className="container"
    data-test-id="Test.png"
    onClick={[MockFunction]}
  >
    <div
      className="innerContainer"
    >
      <div
        className="iconContainer"
      >
        <img
          alt="file"
          className="thumbnail"
          src="https://bejow.solid.community/1234/Test.png"
        />
        <img
          alt="file"
          className="icon"
          src="https://owntech.de/favicon.ico"
        />
      </div>
    </div>
  </div>
  <div
    className="label"
  >
    Test.png
  </div>
</div>
`;

exports[`File should render correctly (no image) 1`] = `
<div
  className="gridCell"
  onContextMenu={[Function]}
>
  <div
    className="container"
    data-test-id="Test.png"
    onClick={[MockFunction]}
  >
    <div
      className="innerContainer"
    >
      <div
        className="iconContainer"
      >
        <img
          alt="file"
          className="thumbnail"
          src="https://bejow.solid.community/1234/Test.png"
        />
        <img
          alt="file"
          className="icon"
          src="https://owntech.de/favicon.ico"
        />
      </div>
    </div>
  </div>
  <div
    className="label"
  >
    Test.png
  </div>
</div>
`;
